data_model/src/error.rs

```diff
@@ -0,0 +1,243 @@
+//! Errors used in Iroha special instructions and
+//! queries. Instruction execution should fail with a specific
+//! error variant, as opposed to a generic (printable-only)
+//! [`eyre::Report`]. If it is impossible to predict what kind of
+//! error shall be raised, there are types that wrap
+//! [`eyre::Report`].
+#![cfg(feature = "std")]
+
+use derive_more::Display;
+use iroha_crypto::HashOf;
+use iroha_primitives::fixed::FixedPointOperationError;
+use iroha_schema::IntoSchema;
+use parity_scale_codec::{Decode, Encode};
+use thiserror::Error;
+
+use crate::{block::VersionedCommittedBlock, metadata, permission, prelude::*, trigger};
```

Comment:
Sure, because client depends on this error structures. `QueryResponseHandler` expects to recieve `QueryError` ([link](https://github.com/hyperledger/iroha/blob/d4f3d85a189ccbe40081188d7f03f67a98282a18/client/src/client.rs#L85)).